Makefile

```
@@ -26,19 +36,24 @@ module_graph.svg: _build/doc/all_modules.docdir/all_modules.dot
 
 # opam1-setup - for running in Wercker. Assumes the correct switch is already installed and selected.
 opam1-setup:
-	opam pin add -y . --no-action
-	opam depext -y fix-engine
-	opam install fix-engine --deps-only
+	opam install \
+	  ./vendor/imandra-ptime/imandra-ptime.opam \
+	  ./vendor/imandra-prelude/imandra-prelude.opam \
+	  ./fix-engine.opam \
+	  --deps-only -y
 
+# Note: keep this opam install command in sync with the Dockerfile
 # opam2-setup - Will create a local switch under ./_opam.
 opam2-setup: _opam
-	opam pin add -y . --no-action
-	opam depext -y fix-engine
-	opam install -y . --deps-only --with-test --working-dir
+	opam install \
+	  ./vendor/imandra-ptime/imandra-ptime.opam \
+	  ./vendor/imandra-prelude/imandra-prelude.opam \
+	  ./fix-engine.opam \
+	  --deps-only -y
 
 _opam:
 	opam switch create . --empty
-	opam install -y ocaml-base-compiler.4.05.0
+	opam install -y ocaml-base-compiler.4.12.1
```

Better to do this instead:
```suggestion
	opam switch set-invariant ocaml-base-compiler.4.12.1
```

That way, the compiler version will be locked in the switch and future opam commands won't cause it to be upgraded/downgraded.

The compiler will be installed when `opam install` is called later.